api/features/feature_segments/tests/test_models.py

```diff
@@ -135,3 +140,22 @@ def test_feature_segments_are_created_with_correct_priority(self):
         assert feature_segment_3.priority == 0
         assert feature_segment_4.priority == 0
         assert feature_segment_5.priority == 0
+
+    def test_clone_creates_a_new_object(self):
+        # Given
+        feature_segment = FeatureSegment.objects.create(
+            feature=self.remote_config,
+            segment=self.segment,
+            environment=self.environment,
+            priority=1,
+        )
+        new_env = Environment.objects.create(
+            name="Test environment New", project=self.project
+        )
+
+        # When
+        feature_segment_clone = feature_segment.clone(new_env)
+
+        # Then
+        assert feature_segment_clone.id != feature_segment.id
+        assert feature_segment_clone.priority == feature_segment.priority
```

Comment:
I'd probably add a check here that the environment on the cloned feature segment is the new clone environment.